src/backend/gporca/libgpopt/src/search/CGroup.cpp

```diff
@@ -1905,6 +1905,12 @@ CGroup::FResetStats()
 		return true;
 	}
 
+	if (FDuplicateGroup())
+	{
+		PgroupDuplicate()->FResetStats();
+		return false;
```

Comment:
What does true or false mean here?

Reply:
Should this be true, to indicate that stats were reset? Might be worth adding a comment too.

Reply:
Usage of calling `FResetStats()` on a group is to reset the group stats if and only if at least one the children group stats were reset.  This return value is used during the recursion to indicate whether a group's stats were reset.  And if they were then it's used to trigger the parent to reset stats too.

Originally we weren't quite sure whether we should reset group stats if child duplicate was reset.  But since `PstatsRecursiveDerive()` can trigger duplicate group to derive stats, we should probably also allow it to reset the duplicate parent stats.

After thinking a bit more, I think following is probably correct:

```suggestion
		return PgroupDuplicate()->FResetStats();
```

Reply:
Ah yeah I think that looks correct

Reply:
+1 to David's suggested change.